lib/temporal/workflow/task_processor.rb

```diff
@@ -40,16 +40,13 @@ def process
 
         complete_task(commands)
       rescue StandardError => error
-        fail_task(error)
-
-        Temporal.logger.error("Workflow task for #{workflow_name} failed with: #{error.inspect}")
-        Temporal.logger.debug(error.backtrace.join("\n"))
-
         Temporal::ErrorHandler.handle(error, metadata: metadata)
+
+        fail_task(error)
```

Comment:
This is a change in behaviour — failing a task after handling errors, can you please elaborate?

Reply:
That's how the workflow one does it. I don't have a preference, but I think it should be consistent.

Reply:
Got it, makes sense, didn't notice these were different to begin with